.github/workflows/build.yml

```diff
@@ -9,34 +9,38 @@ on:
     branches: [ master ]
   pull_request:
     branches: [ master ]
+  schedule:
+    # 3 am Tuesdays and Fridays
+    - cron: "0 3 * * 2,5"
+    
+concurrency:
+  group: "${{ github.ref }}-${{ github.head_ref }}-${{ github.workflow }}"
+  cancel-in-progress: true
 
 jobs:
   build:
-
     runs-on: ${{matrix.os}}
     strategy:
       max-parallel: 12
       fail-fast: false
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
-        python-version: ['3.6', '3.7', '3.8']
+        python-version: ['3.6', '3.7', '3.8', '3.9', '3.10', '3.11']
 
     steps:
     - uses: actions/checkout@v2
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v2
-      with:
-        python-version: ${{ matrix.python-version }}
     - name: Setup Miniconda
-      uses: conda-incubator/setup-miniconda@v2.1.1
+      uses: conda-incubator/setup-miniconda@v2
       with:
          python-version: ${{ matrix.python-version }}
+         auto-update-conda: true
+         channels: conda-forge
+         mamba-version: "*"
+         add-pip-as-python-dependency: true
     - name: Install dependencies
       run: |
-        python -m pip install --upgrade pip
-        python -m pip install  pytest
-        conda install -c anaconda zlib
-        pip install -r requirements.txt
+        mamba install -c conda-forge zlib
+        mamba install --yes --file requirements.txt
```

Comment:
@hmacdope do you mind if I just add a yaml file here to make life easier?

Reply:
Not at all